### ISSUE TITLE
Add a grenades mod

### ISSUE THREAD

**Issue by LoneWolfHT** [MT-CTF/capturetheflag#264]
Would be cool to have a few more grenades. Like a flash bang or smoke grenade. Maybe even a few made up ones.
I've started a mod I'll be releasing on the forums. But it would probably work here too. Thoughts?

**Comment by AKryukov92** [comment]
I don't like proposal about "more of something".
What exactly do you have in mind?

**Comment by LoneWolfHT** [comment]
> I don't like proposal about "more of something".
> What exactly do you have in mind?

Not sure yet. Was thinking people could post suggestions here. Popular ideas are a smoke and flash bang grenade. What I'm making is an API and one basic grenade. More grenades could be added

**Comment by AKryukov92** [comment]
I think people always can post suggestion here. By "here" I mean list of issues in this repository.
Github is not forum. There is no need for "mega" threads about vague ideas.
I think it will be better if you create one issue about flashbang grenade and other - about smoke grenade. Of course these threads should not be empty. You should write why do you propose it and how do you imagine their mechanic in detail.

**Comment by LoneWolfHT** [comment]
I'll close the issue once the API is done or if it gets a lot of negative feedback and create a PR. For now it'll be for suggestions

**Comment by ClobberXD** [comment]
Please don't make one issue for each type of grenade. One issue would suffice.

**Comment by LoneWolfHT** [comment]
>Please don't make one issue for each type of grenade. One issue would suffice.

I'll only have this issue and a PR later for grenades

**Comment by AKryukov92** [comment]
> I'll only have this issue and a PR later for grenades

PRs might not be merged because rubenwardy might not like those features. I think it is better to discuss things first and get approve on idea before doing any work.

**Comment by LoneWolfHT** [comment]
> > I'll only have this issue and a PR later for grenades
> 
> PRs might not be merged because rubenwardy might not like those features. I think it is better to discuss things first and get approve on idea before doing any work.

If PRs won't be merged then said PRs will be changed. This issue is for the discussion of grenades.

**Comment by LoneWolfHT** [comment]
May leave the flashbang until I can sort out the viewing problems. How are the other two grenades though? https://github.com/LoneWolfHT/grenades

**Comment by LoneWolfHT** [comment]
Yeah I was going to start a PR but I had some trouble with getting the game to work. I'm improving the code rn though. So thought I might as well wait until that's finished.

**Comment by LoneWolfHT** [comment]
Maybe test the grenade and the API now. Improved it a little @ClobberXD

**Comment by LoneWolfHT** [comment]
Will try to create a PR today

**Comment by AKryukov92** [comment]
As long as you keep this thread, I'll add one important point.
There are distinction between offensive and devensive grenades.
Main difference - radius of damage. Defensive grenades damage radius is larger, than distance of typical throw. Thats why they are usually dropped behind protective cover.

**Comment by LoneWolfHT** [comment]
I'm not sure what you mean?
